src/array_value.cpp

```diff
@@ -1277,8 +1275,8 @@ ValuePtr ArrayValue::hash(Env *env) {
                 item_hash = item_hash->send(env, to_int);
 
             nat_int_t h = item_hash->as_integer()->to_nat_int_t();
-            hash ^= i;
-            hash ^= h * primes[i % PRIME_COUNT]; // another prime number to prevent bias toward zero
+
+            hash += (hash << 5) ^ h;
```

Comment:
How did you come up with `5` here?

Reply:
any number that would not the number of bits was not divisible by in order to avoid periodicity issues (or really, just delay them by 5 * nat_int_t_bits) that was also not too high and would not completely invalidate the effect of the left-most bits of the previous hash (yeah, sums puts those back, but the unshifted part still had more influence). Another candidate by that train of thought was 3, but then again, 5 is almost twice as much effective in terms of periodicity :) 

Both are conjectures though, I really need to verify how good is the distribution of this :^)

Reply:
OK that makes sense!